### PR TITLE
feat/alert/cleanup : Clean up code from M1 : `ui.alert` package 

### PR DESCRIPTION
**Pull request by charliemangano**
# Clean up code from M1 : ui.alert package 

## Description
This PR introduces code cleanup and new documentation for the files in package `ui.alert`, including corresponding tests. It closes issue #122 and part of #84.

## Changes
**CreateAlert screen and tests:**  

Renamed `AlertScreen.kt` to `CreateAlert.kt` for consistency and alignment with the Figma. Also renamed `AlertScreenTest` to `CreateAlertScreenTest` and `C.Tag.AlertScreen` to `C.Tag.CreateAlertScreen`.  

Extracted relevant values to constant variables.  

Changed `validateFields` return type from `String?` to `Pair<Boolean, String>` for better readability.  

Changed parameters names for better maintainability and modularity, and so that the code is self-documenting.

Added documentation for functions.

Removed useless comments.

**AlertLists screen and tests:**  

Renamed `AlertsList.kt` to `AlertLists.kt` for consistency and alignment with the Figma. Also renamed `AlertsListScreenTest` to `AlertListsScreenTest` and `C.Tag.AlertListScreen` to `C.Tag.AlertListsScreen`.

Extracted relevant values to constant variables.

Added a private `enum` class `AlertListsTab` to represent the two tabs `MY_ALERTS` and `PALS_ALERTS`. Now, instead of having a `selectedTabIndex` with an unnamed, difficult to understand value, we can just pass the `AlertListsTab` around.

Changed variables names for better maintainability and modularity, and so that the code is self-documenting.

Added documentation for functions and TODOs where needed.

#### Added
None
#### Modified
- `app/src/main/java/com/android/periodpals/ui/alert/AlertScreen.kt`: renamed to `CreateAlert.kt`, cleaned up the code and added documentation.
- `app/src/androidTest/java/com/android/periodpals/ui/alert/AlertScreenTest.kt`: renamed to `CreateAlertScreenTest.kt`, cleaned up code
- `app/src/main/java/com/android/periodpals/ui/alert/AlertsList.kt`: renamed to `AlertLists.kt`: cleaned up code and added documentation
- `app/src/androidTest/java/com/android/periodpals/ui/alert/AlertListScreenTest.kt`: renamed to `AlertListsScreenTest.kt`, cleaned up code
- `app/src/main/java/com/android/periodpals/resources/C.kt`: renamed `C.Tag.AlertScreen` to `C.Tag.CreateAlertScreen` and `C.Tag.AlertListScreen` to `C.Tag.AlertListsScreen`

#### Removed
None

## Dependencies Added
None

## Testing
All tests still pass.

### CODE REVIEW

File: app/src/androidTest/java/com/android/periodpals/ui/alert/CreateAlertScreenTest.kt [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+    composeTestRule.onNodeWithTag(CreateAlertScreen.SUBMIT_BUTTON).performClick()
+    verify(navigationActions).navigateTo(Screen.ALERT_LIST)
+  }
+
+  @Test
+  fun createInvalidAlertNoProduct() {
+
+    composeTestRule.onNodeWithTag(CreateAlertScreen.URGENCY_FIELD).performClick()
+    composeTestRule.waitForIdle()
```
Comment (lazarinibruno) [inline app/src/androidTest/java/com/android/periodpals/ui/alert/CreateAlertScreenTest.kt:89]: Question: I'm aware that you just refactored the code, but why is waitForIdle() needed here? I think @agonzalez-r wrote this tests originally, so maybe she knows?

**Review by lazarinibruno — APPROVED** [review]
## Summary

Thanks for the work!

- I appreciate how you extracted the relevant values to constant values, as it improves the readability of the code.
  
- The enum in `AlertLists.kt` to distinguish between both tabs are particularly great for code readability. I believe @agonzalez-r had also proposed it, but I must have forgotten to implement it.
  
- Tests pass locally.
  

## Steps before merge

- None

File: app/src/androidTest/java/com/android/periodpals/ui/alert/CreateAlertScreenTest.kt (hunk 1 above)
Comment (charliemangano) [reply app/src/androidTest/java/com/android/periodpals/ui/alert/CreateAlertScreenTest.kt:89]: i actually don't remember why this was added in but you're right, we should remove it

Comment (agonzalez-r) [reply app/src/androidTest/java/com/android/periodpals/ui/alert/CreateAlertScreenTest.kt:89]: hello, no actually the tests to check if product or urgency fields are not left empty were written by either @francelu or @charliemangano last week, I don't remember whose version we kept
So I'm not sure why it was used :/

**Review by agonzalez-r — APPROVED** [review]
Thank you for this clean up, documentation is very clean. LGTM!!

File: app/src/androidTest/java/com/android/periodpals/ui/alert/CreateAlertScreenTest.kt (hunk 1 above)
Comment (charliemangano) [reply app/src/androidTest/java/com/android/periodpals/ui/alert/CreateAlertScreenTest.kt:89]: I'll get rid of it!

**Pull request merged**